src/async_impl/client.rs

```diff
@@ -2855,12 +2920,15 @@ impl Future for PendingRequest {
                 ResponseFuture::Default(r) => match Pin::new(r).poll(cx) {
                     Poll::Ready(Err(e)) => {
                         #[cfg(feature = "http2")]
-                        if self.as_mut().retry_error(&e) {
-                            continue;
+                        if e.is_request() {
```

Comment:
What's the change here for? Is it because the error was pushing into a `reqwest::Error` type instead of the underlying hyper error? That sounds right, now that I type is out...

Reply:
Yes, in the `HyperService::call` where had wrapped the `crate::error::request` error outside already.
>   Box::pin(async move { inner.call(req).await.map_err(crate::error::request) })